### DISCUSSION TITLE
Can't get map writer plugin for Osmosis to work

### DISCUSSION THREAD

**Discussion by agent457** [mapsforge/mapsforge#1568] [Q&A] [answered]
I have been trying for a while now, following the documentation to the letter.
I downloaded Osmosis through the Debian repositories packages. I then put the most recent .jar of map writer in /usr/share/osmosis/.
but then trying to run it yields nothing, only the error that --mw doesn't exist.
I am new to Linux and development as a whole, so I may be misinterpreting the documents, but I am pretty sure I did exactly as they instructed.
Can anyone please help me with this?

**Comment by devemux86** [comment]
[Map writer guide](https://github.com/mapsforge/mapsforge/blob/master/docs/Getting-Started-Map-Writer.md#plugin-installation) mentions the [Osmosis guide](http://wiki.openstreetmap.org/wiki/Osmosis/Detailed_Usage#Plugin_Tasks) with several ways to install plugins.

I also use Linux and always put the plugins in `~/.openstreetmap/osmosis/plugins`.

We use Osmosis 0.47. Run it in your home folder. It is available in [Osmosis guide](https://wiki.openstreetmap.org/wiki/Osmosis#Downloading).

**Reply by agent457** [reply]
I don't have any dot folder named openstreetmap, and when I tried creating one it did nothing.
Is it because I downloaded using debian repositories package manager?

**Reply by devemux86** [reply, marked as answer]
If a folder does not exist, you can create it.
(it is a hidden folder, enable show hidden files in your file manager)

It is better to download and use an Osmosis zip in your home folder.
(we have currently tested and use version 0.47)

**Reply by agent457** [reply]
thanks a bunch! I got it it work!
Now, it's not necessary (but would be nice), is it possible to do a conversion of a file a user uploads while an app using mapsforge is running?
Like, he selects a .osm map file and I want to use it, is it possible to automatically convert it to .map in the app?

**Reply by devemux86** [reply]
Android or desktop app?
You need to include Osmosis, map-writer and dependencies in the app.

Also large maps take much time and require cpu and memory.
This is not usually work for apps, but command line.

**Reply by agent457** [reply]
Android, and large maps are not a concern, megabyte at most. The app is for small areas.

**Reply by devemux86** [reply]
Creating good offline maps involves more steps and tools than just the osm file.
This is explained with details in the [map creation guide](https://github.com/mapsforge/mapsforge/blob/master/docs/MapCreation.md).

**Reply by agent457** [reply]
I am just trying to make things work currently, (this is just for a finals project in school).
Even if I was concerned with quality, coastlines are not something that will be displayed. The app just displays detailed maps of small area such as schools and malls.
